newIDE/app/src/EventsFunctionsExtensionEditor/index.js

```diff
@@ -578,14 +580,70 @@ export default class EventsFunctionsExtensionEditor extends React.Component<
     );
   };
 
+  /**
+   * Try to find a default value for any required behavior property that is set
+   * with an invalid value.
+   * @param {*} project
+   */
+  _fillRequiredBehaviorProperties = (): boolean => {
+    const project = this.props.project;
+    const problems = gd.WholeProjectRefactorer.findInvalidRequiredBehaviorProperties(
+      project
+    );
+    for (let index = 0; index < problems.size(); index++) {
+      const problem = problems.at(index);
+
+      const object = problem.getSourceObject();
+      const expectedBehaviorTypeName = problem.getExpectedBehaviorTypeName();
+      const suggestedBehaviorNames = gd.WholeProjectRefactorer.getBehaviorsWithType(
+        object,
+        expectedBehaviorTypeName
+      ).toJSArray();
+      const behaviorContent = problem.getSourceBehaviorContent();
+      const behavior = gd.MetadataProvider.getBehaviorMetadata(
+        project.getCurrentPlatform(),
+        behaviorContent.getTypeName()
+      ).get();
+
+      if (suggestedBehaviorNames.length === 0) {
+        // No matching behavior on the object.
+        // Add required behaviors on the object.
+        const defaultName = gd.MetadataProvider.getBehaviorMetadata(
+          project.getCurrentPlatform(),
+          expectedBehaviorTypeName
+        ).getDefaultName();
+        gd.WholeProjectRefactorer.addBehaviorAndRequiredBehaviors(
+          project,
+          object,
+          expectedBehaviorTypeName,
+          defaultName
+        );
+        behavior.updateProperty(
+          behaviorContent.getContent(),
+          problem.getSourcePropertyName(),
+          defaultName
+        );
+      } else {
+        // There is a matching behavior on the object use it by default.
+        behavior.updateProperty(
+          behaviorContent.getContent(),
+          problem.getSourcePropertyName(),
+          // It's unlikely the object has 2 behaviors of the same type.
+          suggestedBehaviorNames[0]
+        );
+      }
+    }
+    return problems.size() > 0;
+  };
+
   _openFreeFunctionsListEditor = () => {
     if (this._editorNavigator)
       this._editorNavigator.openEditor('free-functions-list');
   };
 
   _openBehaviorsListEditor = () => {
     if (this._editorNavigator)
-      this._editorNavigator.openEditor('behaviors-list');
+      return this._editorNavigator.openEditor('behaviors-list');
```

Comment:
Useless return?

Reply:
Yes, sorry about the useless returns I forgot to remove them after putting it on the right method.